### PR TITLE
setting up mongoDB and models

### PR DESCRIPTION
**Pull request by kengneruphine**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -66,6 +66,8 @@
     "dotenv": "^8.2.0",
     "esm": "^3.2.25",
     "regenerator-runtime": "^0.13.3",
-    "rimraf": "^3.0.2"
+    "rimraf": "^3.0.2",
+    "mongoose": "^5.9.2",
+    "body-parser": "^1.19.0"
   }
-}
+}
```
Comment (ibkadeeko) [inline package.json]: Add new line at the end of the file

File: src/app.js [hunk 2]
```diff
@@ -1,13 +1,28 @@
 import express, { json, urlencoded } from 'express';
 import cors from 'cors'
 import morgan from 'morgan'
+import bodyParser from 'body-parser'
+import mongoose from 'mongoose'
+//import route from './routes/index';
 
 const app = express();
+mongoose.connect('mongodb+srv://grace:bIwdzDRydpY28vZc@cluster0-n7fk2.mongodb.net/test?retryWrites=true&w=majority')
```
Comment (ibkadeeko) [inline src/app.js]: Extract this to a config file and put the mongodb url in the environment variables

Comment (ibkadeeko) [inline src/app.js]: Abstract this to a config file and make use of environment variables

File: src/app.js [hunk 3]
```diff
@@ -1,13 +1,28 @@
...
+    .catch((error) => {
+        console.log("Unable to connect to MongoDB ATLAS");
+        console.error(error);
+   })
 
 app.use(cors());
 app.use(morgan('dev'))
 app.use(json());
 app.use(urlencoded({ extended: true }));
+app.use(bodyParser.json());
```
Comment (ibkadeeko) [inline src/app.js]: Why the addition of body Parser.
What does it do?

File: src/app.js [hunk 4]
```diff
@@ -1,13 +1,28 @@
...
+app.use(bodyParser.urlencoded({ extended: true }));
```
Comment (ibkadeeko) [inline src/app.js]: same here

File: src/app.test.js [hunk 5]
```diff
@@ -13,7 +13,7 @@ beforeEach(() => {
 
 describe('Root Route', () => {
   it('should display Welcome on root route', async () => {
-    const res = await request.get('/');
+    const res = await request.get('/'); //requesting our application with a get request on '/'
     res.should.have.status(200);
   });
 });
```
Comment (ibkadeeko) [inline src/app.test.js]: Add new line at the end of the file

File: src/models/list.js [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+
+const listTodoSchema = mongoose.Schema({
+    name: { type: String, required: true },
+    description: { type: String },
+    createdBy: {
+        type: mongoose.Schema.Types.ObjectId,
+        ref: 'User'
+    },
+    // A list has many todos
+    todos: [
```
Comment (ibkadeeko) [inline src/models/list.js]: You would need to restructure your model a little bit. Child and parent must not reference each other. One point is enough

File: src/models/users.js [hunk 7]
```diff
@@ -0,0 +1,19 @@
+import mongoose from 'mongoose'
+
+const userSchema = mongoose.Schema({
+    firstName: { type: String },
+    lastName: { type: String },
+    email: { type: String },
+    password: { type: String },
+    // A user can create many list of todos
+    listTodos: [
```
Comment (ibkadeeko) [inline src/models/users.js]: Go through your models and device a more efficient way to create the relationships between them. Hint: You dont need to reference child and parent.

**Comment by kengneruphine** [comment]
Made the requested changes

File: src/config.js [hunk 8]
```diff
@@ -0,0 +1,3 @@
+const DB = 'mongodb+srv://grace:bIwdzDRydpY28vZc@cluster0-n7fk2.mongodb.net/test?retryWrites=true&w=majority';
+
+export default DB;
```
Comment (ibkadeeko) [inline src/config.js]: This should be saved to a .env file instead

File: src/app.js [hunk 9]
```diff
@@ -1,16 +1,29 @@
...
 
 const app = express();
+mongoose.connect(DB)
+    .then(() => {
+        console.log('Successful connected to MongoDB ATLAS');
+    })
+    .catch((error) => {
+        console.log("Unable to connect to MongoDB ATLAS");
+        console.error(error);
+   })
```
Comment (ibkadeeko) [inline src/app.js]: abstract this to a file of its own and then call it in here, you can ignore that file in your tests to make it pass

File: src/database.js [hunk 10]
```diff
@@ -0,0 +1,17 @@
...
+module.exports = function () {
+    mongoose.connect(DB)
+    .then(() => {
+        console.log('Successful connected to MongoDB ATLAS');
+    })
+    .catch((error) => {
+        console.log("Unable to connect to MongoDB ATLAS");
+        console.error(error);
+   })
+}
```
Comment (ibkadeeko) [inline src/database.js]: Add new line at end of file

**Pull request merged**